modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -105,9 +118,35 @@ public String uri() {
 
     @Override
     public BytesReference content() {
-        return content;
+        final BytesReference contentRef = content;
+        assert contentRef != null && released.get() == false;
+        return contentRef;
     }
 
+    @Override
+    public void release() {
+        if (pooled && released.compareAndSet(false, true)) {
+            request.release();
+        }
+        content = null;
+    }
+
+    @Override
+    public HttpRequest releaseAndCopy() {
+        assert content != null;
```

Comment:
`assert contentRef != null && released.get() == false;`?

Reply:
Also what if `release()` is called concurrently to this?

Reply:
Yea you're right, the mechanics of this are weird ... looking at this again there's no win in `null`ing out the `content` here anyway since the `request` field keeps a reference to the bytes anyway for `Unpooled` buffers and for pooled ones we release the bytes just fine.